flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/spec/JobSpec.java

```diff
@@ -65,4 +65,7 @@ public class JobSpec {
 
     /** Upgrade mode of the Flink job. */
     @EqualsAndHashCode.Exclude private UpgradeMode upgradeMode = UpgradeMode.STATELESS;
+
+    /** Allow checkpoint state that cannot be mapped to any job vertex in tasks. */
+    @EqualsAndHashCode.Exclude private boolean allowNonRestoredState = false;
```

Comment:
I think we should not exclude this from the equals comparison

Reply:
I think this option only affect the next upgrade behavior whether to recover from the unknown state mapping and do not have to trigger a direct upgrade after user change this. So I think it should be same with the upgradeMode.